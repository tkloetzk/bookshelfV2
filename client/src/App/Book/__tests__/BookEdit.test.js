import { MuiThemeProvider } from '@material-ui/core/styles'
import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import { fireEvent, render, wait, cleanup } from '@testing-library/react'
import muiTheme from '../../../config/themeConfig'
import BookEdit from '../BookEdit'
import '@testing-library/jest-dom/extend-expect'

const mockStore = configureMockStore()

describe('Book', () => {
  let book
  let props
  let store

  beforeEach(() => {
    book = {
      categories: ['Toddler', 'Parenting'],
      unread: true,
      owned: true,
      _id: '5c7893eabc9b222a3547548f',
      amazonAverageRating: 4.2,
      amazonRatingsCount: 611,
      price: '',
      isbn: '9780553381436',
      title: 'The Happiest Toddler on the Block',
      subtitle:
        'The New Way to Stop the Daily Battle of Wills and Raise a Secure and Well-Behaved One- to Four-Year-Old',
      description:
        "The pediatrician-author of The Happiest Baby on the Block offers parents a groundbreaking new approach for dealing with toddlers, drawing a parallel between a child's development and humankind's journey to civilization and presenting specific behavioral techniques designed to enhance parent-child communication, alleviate tantrums, and increase positive relationships. Reprint. 60,000 first printing.",
      thumbnail:
        'http://books.google.com/books/content?id=njQpY18IazsC&printsec=frontcover&img=1&zoom=1&source=gbs_api',
      goodreadsAverageRating: 3.5,
      goodreadsRatingsCount: 5451,
      __v: 0,
    }

    store = mockStore({
      bookshelf: { genres: book.categories },
    })

    props = { book, handleSave: jest.fn(), setEditMode: jest.fn() }
  })
  afterEach(cleanup)
  describe('render', () => {
    it('should render as expected', () => {
      const { asFragment } = render(
        <Provider store={store}>
          <MuiThemeProvider theme={muiTheme}>
            <BookEdit {...props} />
          </MuiThemeProvider>
        </Provider>
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })
  describe('handleEditSave', () => {
    it('called when save button clicked', async () => {
      const newTitle = 'new title'
      const { getByTestId, asFragment } = render(
        <Provider store={store}>
          <MuiThemeProvider theme={muiTheme}>
            <BookEdit {...props} />
          </MuiThemeProvider>
        </Provider>
      )

      await wait(() => {
        fireEvent.change(getByTestId('titleTextBar'), {
          target: { id: 'title', value: newTitle },
        })
      })

      await wait(() => {
        fireEvent.click(getByTestId('saveButton'))
      })

      expect(props.handleSave).toHaveBeenCalledWith(
        {
          ...book,
          title: newTitle,
        },
        [{ currentValue: book.title, key: 'title', newValue: newTitle }]
      )
    })
  })

  describe('handleCancel', () => {
    it('called when cancel button clicked', async () => {
      const newTitle = 'new title'
      const { getByTestId, asFragment } = render(
        <Provider store={store}>
          <MuiThemeProvider theme={muiTheme}>
            <BookEdit {...props} />
          </MuiThemeProvider>
        </Provider>
      )

      await wait(() => {
        fireEvent.change(getByTestId('titleTextBar'), {
          target: { id: 'title', value: newTitle },
        })
      })

      await wait(() => {
        fireEvent.click(getByTestId('cancelButton'))
      })
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
