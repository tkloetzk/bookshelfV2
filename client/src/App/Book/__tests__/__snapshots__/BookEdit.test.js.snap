// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book handleCancel called when cancel button clicked 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-535 MuiPaper-rounded"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
      style="height: 100%; overflow-x: scroll;"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiFormLabel-filled"
            data-shrink="true"
            for="title"
          >
            Title
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
              data-testid="titleTextBar"
              id="title"
              style="font-size: 12px;"
              type="text"
              value="The Happiest Toddler on the Block"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
      >
        <label
          class="MuiFormLabel-root makeStyles-text-537"
        >
          Categories
        </label>
        <div
          aria-expanded="false"
          class="MuiAutocomplete-root"
          role="combobox"
        >
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-autoCompleteRoot-536 MuiFormControl-marginDense MuiFormControl-fullWidth"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd MuiInputBase-marginDense"
            >
              <div
                class="MuiChip-root MuiAutocomplete-tag MuiChip-deletable"
                data-tag-index="0"
                role="button"
                tabindex="-1"
              >
                <span
                  class="MuiChip-label"
                >
                  Toddler
                </span>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiChip-deleteIcon"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                  />
                </svg>
              </div>
              <div
                class="MuiChip-root MuiAutocomplete-tag MuiChip-deletable"
                data-tag-index="1"
                role="button"
                tabindex="-1"
              >
                <span
                  class="MuiChip-label"
                >
                  Parenting
                </span>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiChip-deleteIcon"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                  />
                </svg>
              </div>
              <input
                aria-autocomplete="list"
                aria-controls="categories-popup"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="disabled"
                autocorrect="off"
                class="MuiInputBase-input MuiInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                id="categories"
                spellcheck="false"
                type="text"
                value=""
              />
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
                tabindex="-1"
                title="Clear"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiFormLabel-filled"
            data-shrink="true"
            for="description"
          >
            Description
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline MuiInputBase-marginDense"
          >
            <textarea
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
              id="description"
              rows="1"
              style="font-size: 12px; height: 0px; overflow: hidden;"
            >
              The pediatrician-author of The Happiest Baby on the Block offers parents a groundbreaking new approach for dealing with toddlers, drawing a parallel between a child's development and humankind's journey to civilization and presenting specific behavioral techniques designed to enhance parent-child communication, alleviate tantrums, and increase positive relationships. Reprint. 60,000 first printing.
            </textarea>
            <textarea
              aria-hidden="true"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
              readonly=""
              style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; font-size: 12px; width: 100%;"
              tabindex="-1"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiFormLabel-filled"
            data-shrink="true"
            for="amazonAverageRating"
          >
            AZ Rating
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
              id="amazonAverageRating"
              step="0.1"
              style="font-size: 12px;"
              type="number"
              value="4.2"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiFormLabel-filled"
            data-shrink="true"
            for="goodreadsAverageRating"
          >
            GR Rating
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
              id="goodreadsAverageRating"
              step="0.1"
              style="font-size: 12px;"
              type="number"
              value="3.5"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiFormLabel-filled"
            data-shrink="true"
            for="amazonRatingsCount"
          >
            AZ Reviews
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
              id="amazonRatingsCount"
              style="font-size: 12px;"
              type="number"
              value="611"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiFormLabel-filled"
            data-shrink="true"
            for="goodreadsRatingsCount"
          >
            GR Reviews
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
              id="goodreadsRatingsCount"
              style="font-size: 12px;"
              type="number"
              value="5451"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-end MuiGrid-justify-xs-flex-end"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            data-testid="saveButton"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm3-10H5V5h10v4z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            data-testid="cancelButton"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Book render should render as expected 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-1 MuiPaper-rounded"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
      style="height: 100%; overflow-x: scroll;"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiFormLabel-filled"
            data-shrink="true"
            for="title"
          >
            Title
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
              data-testid="titleTextBar"
              id="title"
              style="font-size: 12px;"
              type="text"
              value="The Happiest Toddler on the Block"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
      >
        <label
          class="MuiFormLabel-root makeStyles-text-3"
        >
          Categories
        </label>
        <div
          aria-expanded="false"
          class="MuiAutocomplete-root"
          role="combobox"
        >
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-autoCompleteRoot-2 MuiFormControl-marginDense MuiFormControl-fullWidth"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd MuiInputBase-marginDense"
            >
              <div
                class="MuiChip-root MuiAutocomplete-tag MuiChip-deletable"
                data-tag-index="0"
                role="button"
                tabindex="-1"
              >
                <span
                  class="MuiChip-label"
                >
                  Toddler
                </span>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiChip-deleteIcon"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                  />
                </svg>
              </div>
              <div
                class="MuiChip-root MuiAutocomplete-tag MuiChip-deletable"
                data-tag-index="1"
                role="button"
                tabindex="-1"
              >
                <span
                  class="MuiChip-label"
                >
                  Parenting
                </span>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiChip-deleteIcon"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                  />
                </svg>
              </div>
              <input
                aria-autocomplete="list"
                aria-controls="categories-popup"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="disabled"
                autocorrect="off"
                class="MuiInputBase-input MuiInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                id="categories"
                spellcheck="false"
                type="text"
                value=""
              />
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
                tabindex="-1"
                title="Clear"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiFormLabel-filled"
            data-shrink="true"
            for="description"
          >
            Description
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline MuiInputBase-marginDense"
          >
            <textarea
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
              id="description"
              rows="1"
              style="font-size: 12px; height: 0px; overflow: hidden;"
            >
              The pediatrician-author of The Happiest Baby on the Block offers parents a groundbreaking new approach for dealing with toddlers, drawing a parallel between a child's development and humankind's journey to civilization and presenting specific behavioral techniques designed to enhance parent-child communication, alleviate tantrums, and increase positive relationships. Reprint. 60,000 first printing.
            </textarea>
            <textarea
              aria-hidden="true"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
              readonly=""
              style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; font-size: 12px; width: 100%;"
              tabindex="-1"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiFormLabel-filled"
            data-shrink="true"
            for="amazonAverageRating"
          >
            AZ Rating
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
              id="amazonAverageRating"
              step="0.1"
              style="font-size: 12px;"
              type="number"
              value="4.2"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiFormLabel-filled"
            data-shrink="true"
            for="goodreadsAverageRating"
          >
            GR Rating
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
              id="goodreadsAverageRating"
              step="0.1"
              style="font-size: 12px;"
              type="number"
              value="3.5"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiFormLabel-filled"
            data-shrink="true"
            for="amazonRatingsCount"
          >
            AZ Reviews
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
              id="amazonRatingsCount"
              style="font-size: 12px;"
              type="number"
              value="611"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiFormLabel-filled"
            data-shrink="true"
            for="goodreadsRatingsCount"
          >
            GR Reviews
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
              id="goodreadsRatingsCount"
              style="font-size: 12px;"
              type="number"
              value="5451"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-end MuiGrid-justify-xs-flex-end"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            data-testid="saveButton"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm3-10H5V5h10v4z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            data-testid="cancelButton"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
